opsimulator/opsimulator.go

```diff
@@ -64,19 +63,13 @@ type OpSimulator struct {
 func New(log log.Logger, closeApp context.CancelCauseFunc, port uint64, host string, l1Chain, l2Chain config.Chain, peers map[uint64]config.Chain, interopDelay uint64) *OpSimulator {
 	bgTasksCtx, bgTasksCancel := context.WithCancel(context.Background())
 
-	crossL2Inbox, err := bindings.NewCrossL2Inbox(predeploys.CrossL2InboxAddr, l2Chain.EthClient())
```

Comment:
this does not instantiate `crossL2Inbox` properly because `l2Chain.EthClient()` has not been initialized yet.